orca-kayenta/src/main/groovy/com/netflix/spinnaker/orca/kayenta/pipeline/RunCanaryPipelineStage.groovy

```diff
@@ -40,4 +48,20 @@ class RunCanaryPipelineStage implements StageDefinitionBuilder {
   String getType() {
     STAGE_TYPE
   }
+
+  @Override
+  CancellableStage.Result cancel(Stage stage) {
+    Map<String, Object> context = stage.getContext()
+    String canaryPipelineExecutionId = (String)context.get("canaryPipelineExecutionId")
+
+    log.info("Cancelling stage (stageId: $stage.id: executionId: $stage.execution.id, canaryPipelineExecutionId: $canaryPipelineExecutionId, context: $stage.context)")
+
+    try {
+      kayentaService.cancelPipelineExecution(canaryPipelineExecutionId, "")
+    } catch (Exception e) {
+      log.info("Failed to cancel stage (stageId: $stage.id, executionId: $stage.execution.id), e: $e.message")
```

Comment:
Probably worth `log.error` and including the actual exception vs just the message

Reply:
Makes sense; I'll do that. Btw, this line is verbatim from `JenkinsStage` and `PipelineStage`, so we may want to make the same change there.

Reply:
That's fair.

Reply:
Updated as discussed, and changed those other 2 stages as well.